common4j/src/main/com/microsoft/identity/common/java/exception/BaseException.java

```diff
@@ -189,12 +195,26 @@ public void setUsername(@NonNull final String username) {
         this.mUsername = username;
     }
 
-    public String getExceptionName(){
+    public String getExceptionName() {
         return sName;
     }
 
-    public boolean isCacheable(){
+    public boolean isCacheable() {
         //TODO : ADO 1373343 Add the whole transient exception category.
         return !nonCacheableErrorCodes.contains(mErrorCode);
     }
+
+    /**
+     * Set the telemetry on base exception.
+     *
+     * @param telemetry the {@link List<Map<String, String>>} containing telemetry data
+     */
+    public void setTelemetry(@NonNull final List<Map<String, String>> telemetry) {
```

Comment:
Two things... take this on the constructor, and make a safe-copy.

Reply:
Changed to `add`. Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1636/commits/4f18eff7cf4b93131c2ebd4ec44908688f0efe0b